asaps/models.py

```
@@ -25,36 +27,55 @@ def __init__(self, secfile):
         authclient.authorize()
         self.authclient = authclient
 
-    def getrecord(self, uri):
+    def getrec(self, uri):
         """Retrieve an individual record."""
-        record = self.authclient.get(uri).json()
+        rec = self.authclient.get(uri).json()
         print(uri)
-        recType = record['jsonmodel_type']
-
-        # record type
+        recType = rec['jsonmodel_type']
         if recType == 'resource':
-            fields = [f(field) for field in attr.fields(Resource)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(Resource, rec)
         elif recType == 'accession':
-            fields = [f(field) for field in attr.fields(Accession)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(Accession, rec)
         elif recType == 'archival_object':
-            fields = [f(field) for field in attr.fields(ArchivalObject)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(ArchivalObject, rec)
         else:
-            print('Invalid record type')  # likely better ways of handling this
+            print('Invalid record type')
             exit()
```

The better thing to do here would probably be to raise an exception. At this low level in the application, we don't want to be making decisions about whether to exit the app or not as that's a concern for further up in the app. We can just raise a generic exception for now and later on we can come back and create our own exceptions when we have a better idea of what we'll need:
```python
else:
    raise Exception("Invalid record type")
```